HousingSearchApi.Tests/V1/Helper/TestDataHelper.cs

```diff
@@ -1,38 +1,64 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
 using AutoFixture;
-using HousingSearchApi.V1.Gateways;
 using HousingSearchApi.V1.Gateways.Models;
 using Nest;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
 
 namespace HousingSearchApi.Tests.V1.Helper
 {
     public static class TestDataHelper
     {
+        public static readonly string Index = "persons";
         public static string[] Alphabet = { "aa", "bb", "cc", "dd", "ee", "vv", "ww", "xx", "yy", "zz" };
 
-        public static void InsertPersonInEs(IElasticClient elasticClient)
+        public static List<QueryablePerson> InsertPersonsInEs(IElasticClient elasticClient)
         {
-            elasticClient?.Indices.Delete("persons");
+            elasticClient.Indices.Delete(Index);
 
-            elasticClient?.Indices.Create("persons", s =>
+            elasticClient.Indices.Create(Index, s =>
                 s.Map(x => x.AutoMap()
                     .Properties(prop =>
                         prop.Keyword(field => field.Name("surname"))
                             .Keyword(field => field.Name("firstname")))));
 
-            elasticClient.IndexManyAsync(CreateQueryablePerson(), "persons");
+            var persons = CreateQueryablePerson();
+            elasticClient.IndexManyAsync(persons, Index);
+
+            var timeout = DateTime.UtcNow.AddSeconds(10); // 10 second timeout (make configurable?)
+            while (DateTime.UtcNow < timeout)
+            {
+                var count = elasticClient.Cluster.Stats().Indices.Documents.Count;
+                if (count >= persons.Count)
+                    break;
+
+                Thread.Sleep(200);
+            }
+
+            return persons;
         }
 
         private static List<QueryablePerson> CreateQueryablePerson()
         {
             var listOfPersons = new List<QueryablePerson>();
             var random = new Random();
+            var fixture = new Fixture();
+
+            // Make sure there are 10 of each surname first
```

Comment:
Why ten though?

Reply:
Well the page size test uses 5 so it was just making sure that there would always be more than that.